play-media-native/screens/CreateMedia.tsx

```diff
@@ -0,0 +1,212 @@
+
+import { publishMediaItem } from '../api/queries/mediaItems';
+import { uploadSingleImage } from '../api/queries/uploadMedia';
+import { BottomActions } from '../components/BottomActions/BottomActions';
+import { InputText } from '../components/InputText/InputText';
+import { Toast } from '../components/Toast/Toast';
+import { KeyboardAwareScreen } from '../features/Screen/KeyboardAwareScreen';
+import { getFileType } from '../helpers/media';
+import { useMediaQuery } from '../hooks/useMediaQuery/useMediaQuery';
+import { Media } from '../interfaces/media';
+import { styles } from '../theme/styles';
+import { theme } from '../theme/theme';
+
+export const CreateMediaScreen = ({ navigation, route }) => {
+  const [createdImage, setCreatedImage] = useState<Media>();
+
+  const [isValidating, setIsValidating] = useState(false);
+  const [showSuccessToast, setShowSuccessToast] = useState(false);
+  const [showErrorToast, setShowErrorToast] = useState(false);
+  const [shouldShowBottomActions, setShouldShowBottomActions] = useState(true);
+
+  const { refetch: refetchMediaListing } = useMediaQuery();
+
+  const onNameChange = useCallback((text: string) => {
+    setCreatedImage((prev) => ({
+      ...prev,
+      name: text,
+    }));
+  }, []);
+
+  const onDescriptionChange = useCallback((text: string) => {
+    setCreatedImage((prev) => ({
+      ...prev,
+      description: text,
+    }));
+  }, []);
+
+  const handlePublish = useCallback(async () => {
+    setIsValidating(true);
+
+    await uploadSingleImage({ ...createdImage, stateField: '', stateId: '', uploadStatus: '' })
+      .then(async (uploadedImage) => {
+        await publishMediaItem(uploadedImage.id)
+          .then(async () => {
+            setShowSuccessToast(true);
+            await refetchMediaListing();
+            setIsValidating(false);
+            navigation.navigate('MainTabs');
+          })
+          .catch(() => {
+            setShowErrorToast(true);
+            setIsValidating(false);
+          });
+      })
+      .catch(() => {
+        setShowErrorToast(true);
+        setIsValidating(false);
+      });
+  }, [createdImage, navigation, refetchMediaListing]);
+
+  const handleSaveDraft = useCallback(async () => {
+    setIsValidating(true);
+
+    await uploadSingleImage({ ...createdImage, stateField: '', stateId: '', uploadStatus: '' })
+      .then(async () => {
+        setShowSuccessToast(true);
+        await refetchMediaListing();
+        setIsValidating(false);
+        navigation.navigate('MainTabs');
+      })
+      .catch(() => {
+        setShowErrorToast(true);
+        setIsValidating(false);
+      });
+  }, [createdImage, navigation, refetchMediaListing]);
+
+  useFocusEffect(
+    useCallback(() => {
+      setCreatedImage(
+        route?.params?.image
+          ? {
+              ...route.params.image,
+              description: route.params.image.description || '',
+              name: route.params.image.name || '',
```

Comment:
If we click the back button and then click continue on the DiscardChanges screen, we lose the changes in title & description fields.